build_wheels.py

```diff
@@ -0,0 +1,113 @@
+import json
+
+import requests
+from packaging.requirements import InvalidRequirement
+from packaging.requirements import Requirement
+
+idf_constraints_url = 'https://dl.espressif.com/dl/esp-idf/espidf.constraints.'
+
+# --- Global variables for branches and constraints files to be downloaded requirements for --- #
+# IDF branches are automatically fetched from github, every release branch higher or equal than 5.0 and master is taken
+# For constraints file the version is resolved from IDF release branch
+# The constraints file for master branch is considered to be the latest release version
+# incremented by 1 in minor field (e.g when last release is v5.2 the constrains file for master is v5.3)
+# if there is no such a version, the last release branch constraints file will be used
+# This can be overwritten by setting the variable 'AUTO_VERSION_CONSTRAINS_FILE' to False
+# and append into the list (in if statement bellow) desired version for master branch
+
+AUTO_VERSION_CONSTRAINS_FILE = True
+
+# ESP-IDF branches list
+IDF_BRANCHES_URL = 'https://api.github.com/repos/espressif/esp-idf/branches?protected=true&per_page=100'
+r = requests.get(IDF_BRANCHES_URL)
+if r.status_code == 200:
+    idf_repo_branches = [branch['name'] for branch in r.json()]
+else:
+    print('Failed to fetch ESP-IDF branches.', '\n', r.text)
+    exit()
+
+idf_branches = []
+for branch in idf_repo_branches:
+    if branch.startswith('release'):
+        if float(branch.split('v')[1]) >= 5.0:
+            idf_branches.append(branch)
+idf_branches.append('master')
+
+# Constraints files versions list
+idf_constraints = []
+for branch in idf_branches:
+    try:
+        splitted_branch = branch.split('release/')
+        idf_constraints.append(splitted_branch[1])
+    except IndexError:
+        if AUTO_VERSION_CONSTRAINS_FILE:
```

Comment:
Why do we have a variable for this? Do you plan to overwrite it with and environment variable?

Reply:
Yes, as I thought about something like this as I mentioned above with "hard-coded 6.0".

And speaking of that for the `'6.0' ` in `idf_constraints.append('6.0')` could be a variable too.